### DISCUSSION TITLE
Suggestions for how to manage a relationship

### DISCUSSION THREAD

**Discussion by lsmith212a** [pimcore/pimcore#15616] [Q&A / Support]
Novice user here. Within the out of the box features, can I create a relationship between 2 objects using an attribute of those objects? For example, lets say I have an object for "Product" and an object for "Style". I want to associate multiple products with a particular style, and multiple styles to each product. But there are a lot of products and we regularly add new ones. I'd like to create a relationship that says something like "This style relates to all products with attribute Type = X" or "this style relates to all products within a specific folder" or something along those lines. What I don't want is that when a new product is added that it has to be individually manually added to multiple styles. I'd like to say that when I create a new X type product, it is automatically related to all styles that are related to X type products. Does that make sense? I'm hoping this is low hanging fruit that I'm just missing because I'm novice. Or maybe someone has another idea for how to approach this?

**Comment by plixus** [comment]
Hi @lsmith212a, I hope I understand you correctly; would it perhaps be a solution to add a "StyleCollection"  class with a relation to your styles. Then you can create objects (e.g. StyleCollectionProductTypeX) and add all related styles. On your product Class you can then add the "StyleCollection"-Relation.

**Comment by aryaantony92** [comment]
Hi @lsmith212a You can use relation fields in the objects for creating the relationship. And you can add the relation when you add a new product using the event listeners. You can define some event listeners on post add event to add the relationship. You can find the available list of events here https://pimcore.com/docs/platform/Pimcore/Extending_Pimcore/Event_API_and_Event_Manager#general. and Data Object Events are listed here https://github.com/pimcore/pimcore/blob/11.x/lib/Event/DataObjectEvents.php
